src/apple/macos/disk.rs

```diff
@@ -1,200 +1,120 @@
-                    } else {
-                        // We just assume by default that this is a HDD
-                        DiskType::HDD
-                    }
-                } else {
-                    DiskType::Unknown(-1)
-                };
+    // Removes `/dev/` from the value.
+    let bsd_name = unsafe {
+        CStr::from_ptr(disk.f_mntfromname.as_ptr())
+            .to_bytes()
+            .strip_prefix(b"/dev/")
+            .expect("device mount point in unknown format")
+    };
+
+    let matching =
+        unsafe { ffi::IOBSDNameMatching(ffi::kIOMasterPortDefault, 0, bsd_name.as_ptr().cast()) };
 
-                new_disk(name, mount_point, type_, removable || ejectable)
-            })
-            .collect::<Vec<_>>()
+    if matching.is_null() {
+        return UNKNOWN_DISK_TYPE;
     }
-}
 
-unsafe fn get_dict_value<T, F: FnOnce(*const c_void) -> Option<T>>(
-    dict: CFDictionaryRef,
-    key: &[u8],
-    callback: F,
-) -> Option<T> {
-    match CFReleaser::new(ffi::CFStringCreateWithCStringNoCopy(
-        ptr::null_mut(),
-        key.as_ptr() as *const c_char,
-        cfs::kCFStringEncodingUTF8,
-        kCFAllocatorNull as _,
-    )) {
-        Some(c_key) => {
-            let mut value = std::ptr::null();
-            if CFDictionaryGetValueIfPresent(dict, c_key.inner() as _, &mut value) != 0 {
-                callback(value)
-            } else {
-                None
-            }
-        }
-        None => None,
+    let mut service_iterator: ffi::io_iterator_t = 0;
+
+    if unsafe {
+        ffi::IOServiceGetMatchingServices(
+            ffi::kIOMasterPortDefault,
+            matching.cast(),
+            &mut service_iterator,
+        )
+    } != libc::KERN_SUCCESS
+    {
+        return UNKNOWN_DISK_TYPE;
     }
-}
 
-unsafe fn get_str_value(dict: CFDictionaryRef, key: &[u8]) -> Option<String> {
-    get_dict_value(dict, key, |v| {
-        let v = v as cfs::CFStringRef;
-
-        let len_utf16 = cfs::CFStringGetLength(v);
-        let len_bytes = len_utf16 as usize * 2; // Two bytes per UTF-16 codepoint.
-
-        let v_ptr = cfs::CFStringGetCStringPtr(v, cfs::kCFStringEncodingUTF8);
-        if v_ptr.is_null() {
-            // Fallback on CFStringGetString to read the underlying bytes from the CFString.
-            let mut buf = vec![0; len_bytes];
-            let success = cfs::CFStringGetCString(
-                v,
-                buf.as_mut_ptr(),
-                len_bytes as _,
-                cfs::kCFStringEncodingUTF8,
-            );
-
-            if success != 0 {
-                utils::vec_to_rust(buf)
-            } else {
-                None
+    let service_iterator = IOReleaser::new(service_iterator).unwrap();
+
+    let mut parent_entry: ffi::io_registry_entry_t = 0;
+
+    while let Some(mut current_service_entry) =
+        IOReleaser::new(unsafe { ffi::IOIteratorNext(service_iterator.inner()) })
+    {
+        loop {
+            let result = unsafe {
```

Comment:
What's the point of storing into `result` since it's only used for a `if` comparison?